### PR TITLE
Simplify cmake code

### PR DESCRIPTION
**Pull request by cyyever**
We use various newer CMake features to simplify build system:
1.Caffe2::threads is replaced by threads::threads.
2.Some unused MSVC flags are removed.

cc @malfet @seemethere

### CODE REVIEW

**Review by malfet — APPROVED** [review]
LGTM if it passes binary builds

**Review by malfet — CHANGES_REQUESTED** [review]
Looks like you need to propagate changes you are making to functorch, see numerous binary build failures. Once it's done, please request re-review

File: cmake/MiscCheck.cmake
Comment (malfet) [inline cmake/MiscCheck.cmake:152]: This code is ancient and probably should go, but have you checked if it actually suppresses some errors/warnings? Can you compare size of build log before/after the change?

**Review by malfet — APPROVED** [review]
Thank you for the changes. Please fix lint before landing though.

File: cmake/MiscCheck.cmake
Comment (cyyever) [reply cmake/MiscCheck.cmake:152]: I find MSVC flags set in multiple locations. Some are redundant. I will recheck them when the tests finish.

**Comment by cyyever** [comment]
@malfet  I simplify some logic of setup.py by moving all python3 detection code to CMake part. Do you think this change is suitable?

**Review by Skylion007 — APPROVED** [review]
Looks good to me.

**Pull request closed without merging**
